server/src/main/java/io/spine/server/projection/ProjectionEventDelivery.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.projection;
+
+import io.spine.annotation.SPI;
+import io.spine.core.EventEnvelope;
+import io.spine.server.delivery.EndpointDelivery;
+
+/**
+ * A strategy on delivering the events to the instances of a certain projection type.
+ *
+ * @param <I> the ID type of projection, to which events are being delivered
+ * @param <P> the type of projection
+ * @author Alexander Yevsyukov
+ */
+@SPI
+public abstract class ProjectionEventDelivery<I, P extends Projection<I, ?, ?>>
```

Comment:
@alexander-yevsyukov Do you want me to write tests for this delivery? That's easier for me, probably.